src/OpenTelemetry/Internal/SelfDiagnosticsConfigParser.cs

```diff
@@ -0,0 +1,116 @@
+// <copyright file="SelfDiagnosticsConfigParser.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics.Tracing;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+
+namespace OpenTelemetry.Internal
+{
+    internal class SelfDiagnosticsConfigParser
+    {
+        private const string ConfigFileName = "DiagnosticsConfiguration.json";
+        private const int FileSizeLowerLimit = 1024;  // Lower limit for log file size in KB: 1MB
+        private const int FileSizeUpperLimit = 128 * 1024;  // Upper limit for log file size in KB: 128MB
+
+        /// <summary>
+        /// ConfigBufferSize is the maximum bytes of config file that will be read.
+        /// </summary>
+        private const int ConfigBufferSize = 4 * 1024;
+
+        // This class is called in SelfDiagnosticsConfigRefresher.UpdateMemoryMappedFileFromConfiguration
+        // in both main thread and the worker thread.
+        // In theory the variable won't be access at the same time because worker thread first Task.Delay for a few seconds.
+        private readonly ThreadLocal<byte[]> configBuffer = new ThreadLocal<byte[]>(() => null);
```

Comment:
Do we need a thread-local here or it might be an overkill?

Reply:
I had the same hesitate when I added the comment, now I think it's for sure it's overkill, because even if there is a race condition and failed the first time it gets called in the main thread, it will be self-corrected the next try in 3 seconds.

Reply:
It's an overkill. Removed it.